src/recon.rs

```diff
@@ -4828,3 +4858,301 @@ pub(crate) unsafe fn rav1d_backup_ipred_edge<BD: BitDepth>(
         }
     }
 }
+
+pub(crate) unsafe fn rav1d_copy_pal_block_y<BD: BitDepth>(
+    t: &mut Rav1dTaskContext,
+    f: &mut Rav1dFrameData,
+    bx4: c_int,
+    by4: c_int,
+    bw4: c_int,
+    bh4: c_int,
+) {
+    let pal = if t.frame_thread.pass != 0 {
+        let index = ((t.by >> 1) + (t.bx & 1)) as isize * (f.b4_stride >> 1)
+            + ((t.bx >> 1) + (t.by & 1)) as isize;
+        &f.frame_thread.pal_as_slice::<BD>()[index as usize][0]
+    } else {
+        let interintra_edge_pal =
+            BD::select_mut(&mut t.scratch.c2rust_unnamed_0.interintra_edge_pal);
+        &interintra_edge_pal.0.pal[0]
+    };
+    for al_pal in &mut BD::select_mut(&mut t.al_pal)[0][bx4 as usize..][..bw4 as usize] {
```

Comment:
```suggestion
    for al_pal in BD::select_mut(&mut t.al_pal)[0][bx4 as usize..][..bw4 as usize] {
```

I think the `&mut` is redundant here, since `BD::select_mut` already returns a `&mut`.

Reply:
Removing `&mut` doesn't seem to work here.